tests/integration/models/release.spec.coffee

```diff
@@ -60,6 +61,55 @@ describe 'Release Model', ->
 				promise = balena.models.release.getAllByApplication(999999)
 				m.chai.expect(promise).to.be.rejectedWith('Application not found: 999999')
 
+		describe 'balena.models.release.createFromUrl()', ->
+			# There is a CORS header that only allows us to run this from dashboard.balena-cloud.com
+			return if IS_BROWSER
+
+			TEST_SOURCE_URL = 'https://github.com/balena-io-projects/simple-server-node/archive/v1.0.0.tar.gz'
+
+			it 'should be rejected if the application name does not exist', ->
+				promise = balena.models.release.createFromUrl('HelloWorldApp', { url: TEST_SOURCE_URL })
+				m.chai.expect(promise).to.be.rejectedWith('Application not found: HelloWorldApp')
+
+			it 'should be rejected if the application id does not exist', ->
+				promise = balena.models.release.createFromUrl(999999, { url: TEST_SOURCE_URL })
+				m.chai.expect(promise).to.be.rejectedWith('Application not found: 999999')
+
```

Comment:
Maybe add one test where the url is invalid (we already have tests in the builder for such cases, but just for completeness)

Reply:
Great idea :+1: